lib/discourse_theme/cli_commands/rspec.rb

```diff
@@ -0,0 +1,242 @@
+              message: "Creating #{image} Docker container...",
+              stream: verbose,
+            )
+
+            execute(
+              command:
+                "docker exec -u discourse:discourse #{container_name} ruby script/docker_test.rb --no-tests --checkout-ref origin/tests-passed",
+              message: "Checking out latest Discourse source code...",
+              stream: verbose,
+            )
+
+            execute(
+              command:
+                "docker exec -e SKIP_MULTISITE=1 -u discourse:discourse #{container_name} bundle exec rake docker:test:setup",
+              message: "Setting up Discourse test environment...",
+              stream: verbose,
+            )
+
+            execute(
+              command: "docker exec -u discourse:discourse #{container_name} bin/ember-cli --build",
+              message: "Building Ember CLI assets...",
+              stream: verbose,
+            )
+          end
+
+          rspec_envs = []
+
+          if headless
+            container_ip =
+              execute(
+                command:
+                  "docker inspect #{container_name} --format '{{.NetworkSettings.IPAddress}}'",
+              ).chomp("\n")
+
+            service =
+              start_chromedriver(allowed_origin: "host.docker.internal", allowed_ip: container_ip)
+
+            rspec_envs.push(SELENIUM_HEADLESS_ENV)
+            rspec_envs.push("CAPYBARA_SERVER_HOST=0.0.0.0")
+            rspec_envs.push(
+              "CAPYBARA_REMOTE_DRIVER_URL=http://host.docker.internal:#{service.uri.port}",
+            )
+          end
+
+          rspec_envs = rspec_envs.map { |env| "-e #{env}" }.join(" ")
+
+          begin
+            tmp_theme_directory = File.join(DISCOURSE_THEME_TEST_TMP_DIR, theme_directory_name)
+            FileUtils.mkdir_p(tmp_theme_directory) if !Dir.exist?(tmp_theme_directory)
+            FileUtils.cp_r(spec_directory, File.join(tmp_theme_directory))
+
+            execute(
+              command:
+                "docker exec #{rspec_envs} -t -u discourse:discourse #{container_name} bundle exec rspec #{File.join("/tmp", theme_directory_name, spec_path)}".squeeze(
+                  " ",
+                ),
+              stream: true,
+            )
+          ensure
+            FileUtils.rm_rf(File.join(tmp_theme_directory, "/spec"))
+          end
+        end
+
+        def configure_local_directory(settings)
+          return if settings.local_discourse_directory_configured?
+
+          should_configure_local_directory =
+            UI.yes?(
+              "Would you like to configure a local Discourse repository used to run the RSpec tests? If you select 'n', the tests will be run using a Docker container.",
+            )
+
+          if should_configure_local_directory
+            local_discourse_directory =
+              UI.ask("Please enter the path to the local Discourse directory:")
+
+            unless Dir.exist?(local_discourse_directory)
```

Comment:
I wonder if we should have one more check to ensure this is indeed a Discourse directory.

```
            unless File.exist?("#{local_discourse_directory}/lib/discourse.rb")
              raise DiscourseTheme::ThemeError.new "'#{local_discourse_directory} does not look like a Discourse directory"
            end
```

Reply:
This is a good point :+1: I've implemented this in the latest commit.